api/server/server.go

```diff
@@ -445,36 +446,45 @@ func (s *Server) bindHandlers(ctx context.Context) {
 			apps := v1.Group("/apps/:app")
 			apps.Use(appNameCheck)
 
-			apps.GET("", s.handleAppGet)
-			apps.PATCH("", s.handleAppUpdate)
-			apps.DELETE("", s.handleAppDelete)
-
-			apps.GET("/routes", s.handleRouteList)
-			apps.POST("/routes", s.handleRoutesPostPutPatch)
-			apps.GET("/routes/:route", s.handleRouteGet)
-			apps.PATCH("/routes/*route", s.handleRoutesPostPutPatch)
-			apps.PUT("/routes/*route", s.handleRoutesPostPutPatch)
-			apps.DELETE("/routes/*route", s.handleRouteDelete)
-
-			apps.GET("/calls", s.handleCallList)
+			{
+				withAppCheck := apps.Group("")
+				withAppCheck.Use(s.checkAppPresenceByName())
+				withAppCheck.GET("", s.handleAppGetByName)
+				withAppCheck.PATCH("", s.handleAppUpdate)
+				withAppCheck.DELETE("", s.handleAppDelete)
+				withAppCheck.GET("/routes", s.handleRouteList)
+				withAppCheck.GET("/routes/:route", s.handleRouteGet)
+				withAppCheck.PATCH("/routes/*route", s.handleRoutesPatch)
+				withAppCheck.DELETE("/routes/*route", s.handleRouteDelete)
+				withAppCheck.GET("/calls/:call", s.handleCallGet)
+				withAppCheck.GET("/calls/:call/log", s.handleCallLogGet)
+				withAppCheck.GET("/calls", s.handleCallList)
+			}
 
-			apps.GET("/calls/:call", s.handleCallGet)
-			apps.GET("/calls/:call/log", s.handleCallLogGet)
+			apps.POST("/routes", s.handleRoutesPostPut)
+			apps.PUT("/routes/*route", s.handleRoutesPostPut)
 		}
 
 		{
-			runner := v1.Group("/runner")
+			runner := clean.Group("/runner")
 			runner.PUT("/async", s.handleRunnerEnqueue)
 			runner.GET("/async", s.handleRunnerDequeue)
 
 			runner.POST("/start", s.handleRunnerStart)
 			runner.POST("/finish", s.handleRunnerFinish)
+
+			appsAPIV2 := runner.Group("/apps/:app")
+			appsAPIV2.Use(setAppNameInCtx)
+			appsAPIV2.Use(s.checkAppPresenceByID())
```

Comment:
won't this implicitly be the case for both of the below endpoints? if an app doesn't exist under a certain id, it doesn't exist (i.e. we don't need the check in front of it since an id is supplied directly). same goes for route under an app id.

Reply:
fixed, but had to adjust few other route handlers as well